src/ophys_etl/modules/roi_cell_classifier/compute_classifier_artifacts.py

```diff
@@ -0,0 +1,275 @@
+        required=False,
+        defalut=128,
+        description="Size of square cutout.",
+    )
+
+
+class ClassifierArtifactsGenerator(ArgSchemaParser):
+    """Create cutouts from the average, max, correlation projects for each
+    detected ROI. Additionally return the ROI mask. Store outputs as 8bit
+    pngs.
+    """
+
+    default_schema = ClassifierArtifactsInputSchema
+
+    def run(self):
+        self.logger.name = type(self).__name__
+        self.logger.setLevel(self.args["log_level"])
+
+        ophys_etl_commit_sha = os.environ.get("OPHYS_ETL_COMMIT_SHA",
+                                              "local build")
+        self.logger.info(f"OPHYS_ETL_COMMIT_SHA: {ophys_etl_commit_sha}")
+        t0 = time.time()
+
+        out_dir = pathlib.Path(self.args["out_dir"])
+        if not out_dir.exists():
+            out_dir.mkdir()
+        assert out_dir.is_dir()
+
+        video_path = pathlib.Path(self.args["video_path"])
+        assert video_path.is_file()
+
+        exp_id = video_path.name.split("_")[0]
+
+        roi_path = pathlib.Path(self.args["roi_path"])
+        assert roi_path.is_file()
+
+        graph_path = pathlib.Path(self.args["graph_path"])
+        assert graph_path.is_file()
+
+        proj = self.get_max_avg(video_path)
+        self.logger.info("Calculated mean and max images...")
+        avg_img = proj["avg"]
+        max_img = proj["max"]
+        corr_img = graph_to_img(graph_path)
+        self.logger.info("Calculated correlation image...")
+
+        quantiles = (self.args["low_quantile"], self.args["high_quantile"])
+        q0, q1 = np.quantile(max_img, quantiles)
+        max_img = normalize_array(array=max_img,
+                                  lower_cutoff=q0,
+                                  upper_cutoff=q1)
+
+        q0, q1 = np.quantile(avg_img, quantiles)
+        avg_img = normalize_array(array=avg_img,
+                                  lower_cutoff=q0,
+                                  upper_cutoff=q1)
+
+        q0, q1 = np.quantile(corr_img, quantiles)
+        corr_img = normalize_array(array=corr_img,
+                                   lower_cutoff=q0,
+                                   upper_cutoff=q1)
+
+        self.logger.info("Normalized images...")
+
+        with open(roi_path, "rb") as in_file:
+            extract_roi_list = sanitize_extract_roi_list(
+                json.load(in_file))
+
+        self.logger.info("Creating and writing ROI artifacts...")
+        for roi in extract_roi_list:
+            self.write_thumbnails(extract_roi=roi,
+                                  max_img=max_img,
+                                  avg_img=avg_img,
+                                  corr_img=corr_img,
+                                  exp_id=exp_id,
+                                  out_dir=out_dir)
+
+        self.logger.info(f"Created ROI artifacts in {time.time()-t0:.0f} "
+                         "seconds.")
+
+    def get_max_avg(self,
```

Comment:
Should this just be a utility function, rather than a method in this class? I don't think it uses `self` at all.

Reply:
I can move this to a utility function somewhere as it likely would have uses in other pipelines.

Reply:
If you're looking for a home, there's `ophys_etl/utils/video_utils.py`